crates/db_views/src/local_user/local_user_view.rs

```diff
@@ -17,90 +20,112 @@ use lemmy_db_schema::{
     DbPool,
   },
 };
-use lemmy_utils::error::{LemmyError, LemmyErrorExt, LemmyErrorType, LemmyResult};
+use lemmy_utils::error::{LemmyError, LemmyErrorExt2, LemmyErrorType, LemmyResult};
 use std::future::{ready, Ready};
 
 impl LocalUserView {
   #[diesel::dsl::auto_type(no_type_alias)]
-  fn joins() -> _ {
-    local_user::table.inner_join(person::table)
+  fn joins(local_instance_id: InstanceId) -> _ {
+    let p: person_with_instance_actions = person_with_instance_actions(local_instance_id);
+    local_user::table.inner_join(p)
   }
 
-  pub async fn read(pool: &mut DbPool<'_>, local_user_id: LocalUserId) -> Result<Self, Error> {
+  pub async fn read(pool: &mut DbPool<'_>, local_user_id: LocalUserId) -> LemmyResult<Self> {
+    let local_instance_id = SiteView::read_local(pool).await?.instance.id;
```

Comment:
I'd much prefer all these `read_local`s were brought up a level. Instead make `local_instance_id` a function param.

Lets try to keep view and schema functions to a single DB query whenever possible, and keep multiple reads at the API / APUB level.

Reply:
I dont like that idea because it makes all the callsites more verbose. These methods are called in many dozens of places so it would have to be passed everywhere, its much easier to read the instance id only a couple of times here. Anyway thanks to caching it doesnt perform a actual db read in most cases.